### DISCUSSION TITLE
Versions improvements

### DISCUSSION THREAD

**Discussion by rainer-friederich** [payloadcms/payload#7981] [Feature Requests & Ideas]
Versions and autosave are a really good feature. However there are multiple areas where we can improve that for authors:

1. Currently versions do not show the title of the page as title in the breadcrumb but `Pages / [UNTITLED] / Versions / <DATE>`
2. There should be a way to hide all content of the versions that did not change. This should be the default IMHO.
3. The default should be to only show the current selected locale of the locale switcher and not all locales.
4. There should be a way to hide versions in the versions overview which only have changes to meta data like (updated., created, published, status, ...) Perhaps there can be a "version ignore" setting for the collections.

**Comment by tsemachh** [comment]
2  and 4 (Assuming meta is hidden) is pushed as PR:
[https://github.com/payloadcms/payload/pull/8074](https://github.com/payloadcms/payload/pull/8074)
I am waiting for payload team to review it

**Reply by tylkomat** [reply]
- fields should be shown based on access control (only for display, when resetting to another version, all fields should be reset)
- richtext fields should be rendered (not show the json structure)
- only fields with text content should be diffed, other fields should just show the whole content as different, i.e. if the id in a relationship field changes, it does not matter which letter in the id (in case of mongodb) was changed, but the content as a whole, relationships should probably render the use as title content of the relation
- callback functions to allow formatting of the displayed values (dates, arrays as pills, values of custom select components,...)
